imcache.go

```diff
@@ -249,6 +249,38 @@ func (c *Cache[K, V]) getAll(now time.Time) map[K]V {
 	return got
 }
 
+// GetAndRemove returns the value for the given key
+// and delete it even if it is not expired
+func (c *Cache[K, V]) GetAndRemove(key K) (V, bool) {
+	now := nowf()
+	var zero V
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	if c.closed {
+		return zero, false
+	}
+	node, ok := c.m[key]
+	if !ok {
+		return zero, false
+	}
+	entry := node.entry()
+	if entry.expired(now) {
+		c.queue.remove(node)
```

Comment:
If the entry is found, we always remove it. This can be simplified then and the removal logic can be moved outside of the `if` statement and then simply if the eviction callback exists, execute callback with a proper eviction reason. It is almost the same as the `Remove` method but it just returns a value as well, see https://github.com/erni27/imcache/blob/master/imcache.go#L638.

Could you also refactor `Remove` method so it uses `GetAndRemove` under the hood but discards the value?